From: wangzijie <wangzijie1@honor.com>
Subject: proc-fix-missing-pde_set_flags-for-net-proc-files-v3
Date: Thu, 21 Aug 2025 18:58:06 +0800

stash pde->proc_ops in a local const variable, per Christian

Link: https://lkml.kernel.org/r/20250821105806.1453833-1-wangzijie1@honor.com
Fixes: ff7ec8dc1b64 ("proc: use the same treatment to check proc_lseek as ones for proc_read_iter et.al")
Signed-off-by: wangzijie <wangzijie1@honor.com>
Reported-by: Lars Wendler <polynomial-c@gmx.de>
Cc: Alexei Starovoitov <ast@kernel.org>
Cc: Alexey Dobriyan <adobriyan@gmail.com>
Cc: Al Viro <viro@zeniv.linux.org.uk>
Cc: "Edgecombe, Rick P" <rick.p.edgecombe@intel.com>
Cc: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Cc: Jiri Slaby <jirislaby@kernel.org>
Cc: Kirill A. Shutemov <k.shutemov@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/proc/generic.c |   12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

--- a/fs/proc/generic.c~proc-fix-missing-pde_set_flags-for-net-proc-files-v3
+++ a/fs/proc/generic.c
@@ -369,18 +369,20 @@ static const struct inode_operations pro
 
 static void pde_set_flags(struct proc_dir_entry *pde)
 {
-	if (!pde->proc_ops)
+	const struct proc_ops *proc_ops = pde->proc_ops;
+
+	if (!proc_ops)
 		return;
 
-	if (pde->proc_ops->proc_flags & PROC_ENTRY_PERMANENT)
+	if (proc_ops->proc_flags & PROC_ENTRY_PERMANENT)
 		pde->flags |= PROC_ENTRY_PERMANENT;
-	if (pde->proc_ops->proc_read_iter)
+	if (proc_ops->proc_read_iter)
 		pde->flags |= PROC_ENTRY_proc_read_iter;
 #ifdef CONFIG_COMPAT
-	if (pde->proc_ops->proc_compat_ioctl)
+	if (proc_ops->proc_compat_ioctl)
 		pde->flags |= PROC_ENTRY_proc_compat_ioctl;
 #endif
-	if (pde->proc_ops->proc_lseek)
+	if (proc_ops->proc_lseek)
 		pde->flags |= PROC_ENTRY_proc_lseek;
 }
 
_
